src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -1078,12 +1078,10 @@ private static async Task InternalWriteAllLinesAsync(StreamWriter writer, IEnume
             {
                 foreach (string line in contents)
                 {
-                    cancellationToken.ThrowIfCancellationRequested();
-                    await writer.WriteLineAsync(line).ConfigureAwait(false);
+                    await writer.WriteLineAsync(line.AsMemory(), cancellationToken).ConfigureAwait(false);
```

Comment:
Glad we have [CA2016](https://github.com/dotnet/runtime/blob/ba777a8ba43dde51bda84a13e0b251492fb9770a/eng/CodeAnalysis.src.globalconfig#L504-L505) enabled as a warning in the repo. 👌